library/swift/src/StreamEmitter.swift

```diff
@@ -30,6 +30,11 @@ public protocol StreamEmitter: CancelableStream {
   ///
   /// - parameter trailers: Trailers to send over the stream.
   func close(trailers: [String: [String]])
+
+  /// End the stream with one last data frame.
+  ///
+  /// - parameter data: Data to send over the stream.
+  func closeWithData(_ data: Data)
```

Comment:
I feel we could expose the endStream flag on the `sendData` handler instead (for ergonomics, we can have that default to `false`). The gRPC implementation detail would just be that we send an empty data with end stream flag being true on the `close()`

WDYT @junr03 @rebello95 ?

Reply:
@buildbreaker and I discussed this a bit offline and thing changing to `close(data: Data)` here probably makes the most sense:
- Matches `close(trailers: [String: [String]])`
- Returns `void` to make it clear that this is a terminal state (rather than returning a `StreamEmitter` for chaining like `sendData`